src/js/model/oid/oid.js

```diff
@@ -0,0 +1,273 @@
+var AMPParse = AMPParse || {};
+
+/**
+ * Builds an OID object as per the spec.
+ *
+ * @param hexadecimal {AMPHexConsumer} The hex consumer to draw bytes from
+ * @param isCompressed {boolean} Whether or not this is a compressed OID.
+ * @return {Object} The object representation of the OID
+ */
+AMPParse.buildOid = function(hexadecimal, isCompressed)
+{
+	// Defensive programming
+	if ( !(hexadecimal instanceof AMPHexConsumer) )
+	{
+		throw new ReferenceError("Did not receive an AMPHexConsumer");
+	}
+	if (typeof isCompressed !== "boolean")
+	{
+		throw new ReferenceError("Provided isCompressed flag is not a boolean.");
+	}
+
+	var consumedNibbles = 0;
+	var returnValue = { type: "Object Identifier" };
+	var nickname;
+	var heirarchy;
+
+	// Read the nickname for a compressed OID
+	if (isCompressed)
+	{
+		try
+		{
+			nickname = AMPParse.buildSdnv(hexadecimal);
+		}
+		catch (err)
+		{
+			throw err;
+		}
+
+		returnValue.nickname = "[" + nickname.returnValue.value + "]";
+		consumedNibbles = nickname.nibblesConsumed;
+		nickname = nickname.returnValue.value;
+	}
+
+	// Read the blob and map the values into their own array
+	try
+	{
+		heirarchy = AMPParse.buildBlob(hexadecimal);
+	}
+	catch (err)
+	{
+		err.nibblesConsumed += consumedNibbles;
+		throw err;
+	}
+
+	consumedNibbles += heirarchy.nibblesConsumed;
+	heirarchy = heirarchy.returnValue.value.map(function(byteOutput) {
+		return byteOutput.returnValue.value;
+	});
+
+	// Turn the first byte into the first two indices if the OID is uncompressed, otherwise prepend nickname
+	if (!isCompressed)
+	{
+		var firstNumber = heirarchy.shift();
+		var secondNumber = firstNumber % 40;
+		firstNumber = (firstNumber - secondNumber) / 40;
+
+		heirarchy.unshift(firstNumber, secondNumber);
+	}
+
+	// Shallow copy the heirarchy array and prepend the default OID header if it's compressed
+	var lookupHeirarchy = heirarchy.slice();
+
+	if (isCompressed)
+	{
+		lookupHeirarchy.unshift(nickname); // Add the nickname!
+		lookupHeirarchy = _defaultOidRoot.concat(lookupHeirarchy);
+	}
+
```

Comment:
once we know the part of the tree we are going to (i.e. Agent Metadata, Agent Primitive Data ... Agent Operators), should we keep that label as part of the return value or do we not care since we will be showing the more detailed name in the extra info?